Makefile

```diff
@@ -1,122 +1,99 @@
+#            |  RPM packages for Centos7  |           v
+#            | Debian packages for Xenial |    +--------------+
+#            | Debian packages for Trusty |    | calico/felix |
+#            +----------------------------+    +--------------+
+#
+#
 
 help:
 	@echo "Felix Makefile"
 	@echo
-	@echo "Dependencies: docker 1.12+; go 1.7+; python 2.7; tox; virtualenv"
+	@echo "Dependencies: docker 1.12+; go 1.7+"
 	@echo
 	@echo "Note: initial builds can be slow because they generate docker-based"
 	@echo "build environments."
 	@echo
 	@echo "Initial set-up:"
 	@echo
 	@echo "  make update-tools  Update/install the go build dependencies."
-	@echo "  make env           Create a Python virtualenv for UT dev."
 	@echo
 	@echo "Builds:"
 	@echo
 	@echo "  make all           Build all the binary packages."
-	@echo "  make pyinstaller   Build pyinstaller bundle in ./dist."
 	@echo "  make deb           Build debs in ./dist."
 	@echo "  make rpm           Build rpms in ./dist."
 	@echo "  make calico/felix  Build calico/felix docker image."
 	@echo
 	@echo "Tests:"
 	@echo
-	@echo "  make ut            Run all UTs."
-	@echo "  make go-ut         Run go UTs (and coverage)."
-	@echo "  make python-ut     Run Python UTs (and coverage)."
+	@echo "  make ut                Run UTs."
 	@echo "  make go-cover-browser  Display go code coverage in browser."
 	@echo
 	@echo "Maintenance:"
 	@echo
-	@echo "  make update-vendor  Update the go/vendor directory with new "
-	@echo "                     versions ofupstream packages.  Record results"
-	@echo "                     in go/glide.lock."
-	@echo "  make update-frozen-reqs  Update the frozen python requirements."
-	@echo "                     Should be run after revving or adding a new"
-	@echo "                     python dependency."
+	@echo "  make update-vendor  Update the vendor directory with new "
+	@echo "                      versions of upstream packages.  Record results"
+	@echo "                      in glide.lock."
 	@echo "  make go-fmt        Format our go code."
 	@echo "  make clean         Remove binary files."
 
 # Disable make's implicit rules, which are not useful for golang, and slow down the build
 # considerably.
 .SUFFIXES:
 
-all: pyinstaller deb rpm calico/felix
+all: deb rpm calico/felix
 test: ut
 
-# re-define default --compare-branch=origin/master to some custom name
-UT_COMPARE_BRANCH?=
-
-# Generate and return the python package version.
-PY_VERSION:=$(shell utils/gen-version.sh)
-
-# Figure out what git commit we have checked out.  We'll bake that into the
-# executable.
-GIT_COMMIT:=$(shell git rev-parse HEAD)
-GIT_COMMIT_SHORT:=$(shell git rev-parse --short HEAD)
-GIT_DESCRIPTION:=$(shell git describe --tags)
+# Figure out version information.  To support builds from release tarballs, we default to
+# <unknown> if this isn't a git checkout.
+GIT_COMMIT:=$(shell git rev-parse HEAD || echo '<unknown>')
+BUILD_ID:=$(shell git rev-parse HEAD || uuidgen | sed 's/-//g')
```

Comment:
Not sure about operator precedence here.  In the Git case, does  the output go through sed?